### PR TITLE
Add auth to New Product API

### PR DESCRIPTION
**Pull request by michaelbjacobson**
This PR adds an API key and requires it for access to the two New Product API endpoints.

### CODE REVIEW

File: cdk/lib/new-product-api.ts [hunk 1]
```diff
@@ -126,6 +128,30 @@ export class NewProductApi extends GuStack {
...
+      name: `new-product-api-usage-plan-${this.stage}`,
+      apiStages: [
+        {
+          api: newProductApi.api,
+          stage: newProductApi.api.deploymentStage
+        }
+      ]
+    })
+
+    const apiKey = new ApiKey(this, "NewProductApiKey", {
```
Comment (rupertbates) [inline cdk/lib/new-product-api.ts:142]: Is the key from the old stack still around?

**Review by rupertbates — APPROVED** [review]
Looks good 👍

File: cdk/lib/new-product-api.ts (hunk 1 above)
Comment (michaelbjacobson) [reply cdk/lib/new-product-api.ts:142]: Nope, the old keys got deleted during the migration to CDK.

**Comment by johnduffell** [comment]
looks good although very minor point I would have called the commit "add auth to new-product-api" rather than initial commit as it makes it easier to understand the history

**Pull request merged**